Dynatrace/dynatrace-operator
# Create istio exclusion for CSI Driver in case of codeModules or public registry

## Description

With this the CSI Driver is excluded from istio blocking traffic if:
- `codeModulesImage` is set
- `public registry` is enabled

`VirtualService` and `ServiceEntry` is created that points to the host where the image is stored. If this is not the case the injection into a pod would not work as everything got stuck because the CSI Driver could not be able to retrieve the image.

~I placed the reconciliation of the `istioReconciler` right before the injection reconciliation, please give me feedback if you are fine with this.~
I placed the reconciliation after the codeModules versionReconciler is finished.
 Currently I am only checking if the `ImageID` is set in the status of the `CodeModule`, this should cover both cases (`codeModulesImage` set or `public registry` enabled).

Respective ticket: https://dt-rnd.atlassian.net/browse/K8S-9622



## How can this be tested?

Deploy the operator with istio, do not forget to label the to be injected namespace accordingly (`istio-injection:enabled`)
Apply a dynakube with istio enabled and
- with a `codeModulesImage` set
- with `public registry `feature flag enabled

Also check for `VirtualServices` and `ServiceEntries` and look if they are created correctly.
